test/compilable/test18348.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -eu pipefail
+
+dir=${RESULTS_DIR}${SEP}runnable
+output_file=${dir}/test13774.sh.out
+
+faketty () { script -qfc "$(printf "%q " "$@")" /dev/null ; }
+
+check()
+{
+    # the first sed removes remove carriage returns
+    out=$(echo "$1" |
+        faketty "$DMD" -c -o- -color=on - | tail -n+2 | \
+        sed 's/\x0d//g' | \
+        sed 's/\x1b\[m/.empty./g' |
+        sed 's/\x1b\[1m/.reset./g' | \
+        sed 's/\x1b\[\([0-9;]*\)m/_set:\1_/g')
```

Comment:
Sanitizes the output, s.t. it's easy to look at without bumping your head against weird escape codes.